src/orange/OrangeParams.cc

```diff
@@ -125,22 +138,31 @@ OrangeParams::OrangeParams(OrangeInput input)
                       "stack is limited to a depth of "
                    << detail::LogicStack::max_stack_depth());
 
+    std::vector<Label> surface_labels;
+    std::vector<Label> volume_labels;
+
     // TODO: update this to work over multiple universe levels
+    for (const UnitInput& u : input.units)
     {
         // Capture metadata
-        UnitInput& u = input.units.front();
-        surf_labels_ = LabelIdMultiMap<SurfaceId>{std::move(u.surfaces.labels)};
-        std::vector<Label> volume_labels;
-        volume_labels.resize(u.volumes.size());
-        for (auto i : range(u.volumes.size()))
+        for (const auto& sl : u.surfaces.labels)
+        {
+            surface_labels.push_back(u.label.name + std::string(".") + sl.name);
+        }
+
+        for (const auto& v : u.volumes)
         {
-            volume_labels[i] = std::move(u.volumes[i].label);
+            volume_labels.push_back(u.label.name + std::string(".")
+                                    + v.label.name);
```

Comment:
```suggestion
            Label vol_label{v.label.name, u.label.name};
            if (v.label.ext)
            {
                vol_label.ext += "." + v.label.ext;
            }
            volume_labels.push_back(vol_label);
```

Reply:
Same thing for surface labels I pressume?

Reply:
Wait, if aren't using the `UniverseId` to uniqify and we are preserving the `ext` from the input (if it exists) shouldn't we just be doing:
```
        for (const auto& v : u.volumes)
        {
            volume_labels.push_back(v.label);
        }
```

Reply:
(as a copy with an lambda function)

Reply:
I thought we decide to use the universe name to uniquify via the extension if the incoming label lacks an extension? So if the input is `"a@b"` it turns into `{"a", "b"}`, but if it's `"a"` in universe 'u' then it becomes `{"a", "u"}`

Reply:
I think I got turned around on this --- I will change it to the above code. I will do the same for surfaces unless I hear otherwise.